### PR TITLE
Upgrate to Nette\DI 3.0

### PR DESCRIPTION
**Pull request by holantomas**

### CODE REVIEW

File: src/DI/CzechPostExtension.php [hunk 1]
```diff
@@ -27,10 +29,26 @@ class CzechPostExtension extends CompilerExtension
 		],
 	];
 
+	public function getConfigSchema(): Nette\Schema\Schema
```
Comment (f3l1x) [inline src/DI/CzechPostExtension.php]: Import Nette\Schema\Schema namespace.

File: src/DI/CzechPostExtension.php [hunk 2]
```diff
@@ -27,10 +29,26 @@ class CzechPostExtension extends CompilerExtension
...
+	{
+		return Expect::structure(
+			[
+				'http' => Expect::structure([
+					'base_uri' => Expect::string('https://online.postservis.cz/'),
+					'auth' => Expect::array(['', '']),
```
Comment (f3l1x) [inline src/DI/CzechPostExtension.php:38]: Why `auth` key is array with empty values?

File: src/DI/CzechPostExtension.php [hunk 3]
```diff
@@ -27,10 +29,26 @@ class CzechPostExtension extends CompilerExtension
...
+														 ]),
+				'config' => Expect::structure([
+					 'tmp_dir' => Expect::type('string|null')->default(null),
+														 ]),
```
Comment (f3l1x) [inline src/DI/CzechPostExtension.php]: Formatting is broken.

**Review by f3l1x — CHANGES_REQUESTED** [review]

File: src/DI/CzechPostExtension.php (hunk 2 above)
Comment (holantomas) [reply src/DI/CzechPostExtension.php:38]: Don't know ... just used values from old `$defaults` property.

File: src/DI/CzechPostExtension.php (hunk 3 above)
Comment (holantomas) [reply src/DI/CzechPostExtension.php]: I don't see problem, tried run codestyle checker `make cs` and for sure `make csf` without error.

File: src/DI/CzechPostExtension.php (hunk 1 above)
Comment (holantomas) [reply src/DI/CzechPostExtension.php]: Fixed

File: src/DI/CzechPostExtension.php (hunk 3 above)
Comment (f3l1x) [reply src/DI/CzechPostExtension.php]: czech-post_CzechPostExtension_php_at_fd47c566c5db1559d832c8d05517c00ebb39803c_·_contributte_czech-post

Comment (holantomas) [reply src/DI/CzechPostExtension.php]: Hope, now it's ok.

File: src/DI/CzechPostExtension.php (hunk 2 above)
Comment (mabar) [reply src/DI/CzechPostExtension.php:38]: I think it's the format expected by guzzle. I would go for explicit `user` and `password` keys and map it to the `[$config['http']['auth']['user'], $config['http']['auth']['password']]`

```phpp
$res = $client->request('GET', 'https://example.com', [
    'auth' => ['user', 'pass']
]);
```

**Comment by f3l1x** [comment]
Awesome, thx @holantomas.

**Pull request merged**